src/daemon/db_util.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::blocks::Tipset;
+use crate::cli_shared::{
+    chain_path,
+    cli::{CliOpts, Config},
+    snapshot,
+};
+use crate::daemon::asyncify;
+use crate::db::car::forest::FOREST_CAR_FILE_EXTENSION;
+use crate::db::car::{ForestCar, ManyCar};
+use crate::db::db_engine::{db_root, open_proxy_db};
+use crate::db::rolling::RollingDB;
+use crate::shim::{clock::ChainEpoch, version::NetworkVersion};
+use crate::utils::db::car_stream::CarStream;
+use crate::utils::{retry, RetryArgs};
+use anyhow::{bail, Context};
+use dialoguer::theme::ColorfulTheme;
+use futures::TryStreamExt;
+use positioned_io::RandomAccessFile;
+use std::ffi::OsStr;
+use std::fs;
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+    time,
+    time::Duration,
+};
+use tokio::io::AsyncWriteExt;
+use tracing::{error, info, warn};
+use url::Url;
+use walkdir::WalkDir;
+
+/// This function tries to open the forest database directory as [`ManyCar<Arc<RollingDB>>`], it
+/// 1. loads `parity-db`
+/// 2. loads all existing CAR files in `.forest.car.zst` format
+/// 3. asks to fetch the latest snapshot when it's required to run forest.
+/// 4. imports the snapshot(from either CLI options or step 3) and stores it to the database in `.forest.car.zst` format
```

Comment:
I don't see why all of these steps have to be done in a single function. Loading CAR files from disk seems to be completely separate from ParityDB, user prompting, and file downloading. Let's separate it out.

Reply:
The goal of this function is to hide all these steps to `fn start` in `mod.rs`, which could DB with 1-liner

Reply:
Are you suggesting making all these steps a separate function and use them all in `mod.rs` or still having a wrapper fn here in `db_util.rs`?

Reply:
Either should be fine to start with. Once the unrelated tasks have been separated into distinct functions then we can do more refactoring.

Reply:
Fixed